framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -879,71 +877,34 @@ public void typeProcess(TypeElement e, TreePath p) {
     ///
 
     /**
-     * Reports a result. By default, it prints it to the screen via the compiler's internal messager
-     * if the result is non-success; otherwise, the method returns with no side effects.
+     * Reports a diagnostic message. By default, it prints it to the screen via the compiler's
+     * internal messager; otherwise, the method returns with no side effects.
      *
-     * @param r the result to report
-     * @param src the position object associated with the result; may be an Element, a Tree, or null
+     * @param source the source position information; may be an Element, a Tree, or null
+     * @param kind the type of message
+     * @param messageKey the message key
+     * @param args arguments for interpolation in the string corresponding to the given message key
      */
-    public void report(final Result r, final Object src) {
-        if (r.isSuccess()) {
-            return;
-        }
+    public void report(
```

Comment:
What do you think of introducing two methods `reportError` and `reportWarning` and discouraging the use of  the version with a `javax.tools.Diagnostic.Kind` dependency? That would make it easier to use the API, removing the need for many of the imports this PR adds.

Reply:
I considered that and thought this variant a bit cleaner.  But I'll change it to your suggestion.  You are right that it will limit the number of imports needed.

Reply:
I have done this and I agree it's better.  Thanks for the suggestion.
Please go ahead and re-review.